Converts pipeline name to lower case

navidshaikh opened the pull request:
Fix OSIO 1522

 Since OpenShift doesn't allow the value of `metadata.name` to have uppercase chars.

mohammedzee1000 committed:
Adding path for static content.

mohammedzee1000 committed:
Commenting out settings.py addition as deploy should handle it.

Deployment should put in steps to update allowed hosts. However
I was manually updating settings and building for test purposes.

mohammedzee1000 committed:
Adding versioning to the API service.

Added versioning class to use namespace versioning.
Updated serializers to be version based.
Updated views to get their serializer class based on namespace.
Updated urls to reflect the version based system.

mohammedzee1000 committed:
Adding missing namespace parameter in urls.py.

mohammedzee1000 committed:
Removing django managed versioning in favour of manual versioning.

This is due to the breaks in the documentation.

mohammedzee1000 committed:
Adding pagination to the API.

mohammedzee1000 committed:
Updating default and maximum page sizes.

mohammedzee1000 committed:
Implementing own pagination class and setting it as default pagination.

mohammedzee1000 committed:
Updating contaienr install script to make it independent.

mohammedzee1000 committed:
Restructured the way API code is organized to make it cleaner.

mohammedzee1000 committed:
Removing Dockerfile.apiserver and putting a part of it.

Also, updating install script accordingly.

mohammedzee1000 committed:
Restructuring the API code to make it cleaner.

Moved api core code to api directory.
Moved api vi code to api/v1 directory.
The main urls now does not initialize any routes, but takes them from API.

mohammedzee1000 committed:
Moving constants to lib.

mohammedzee1000 committed:
Deleted old api directory.

mohammedzee1000 committed:
Moved API code to new directory.

mohammedzee1000 committed:
Updating urls according to changes.

mohammedzee1000 committed:
Fixed list format for api urls.

mohammedzee1000 committed:
Correcting router urls.

mohammedzee1000 committed:
Updating to simpler return for now as we have one version.

mohammedzee1000 committed:
Adding docstrings.

mohammedzee1000 committed:
Updating endpoint paths.

mohammedzee1000 committed:
Adding tags and removing unnessasary ignore_errors.

mohammedzee1000 committed:
Updated the dockerfile and utils.

mohammedzee1000 committed:
Updating provisioning automating bringing up apiserver

mohammedzee1000 committed:
Adding tests for the API.

mohammedzee1000 committed:
Moving provisioning of API Server to end.

mohammedzee1000 committed:
Fixing volume mount for apiserver container

mohammedzee1000 committed:
Removing db role from api setup.

mohammedzee1000 committed:
Adding function to utils to request json data.

mohammedzee1000 committed:
Adding first tests for API.

mohammedzee1000 committed:
Adding constants to CI.

mohammedzee1000 committed:
Updating to use constants as keys.

mohammedzee1000 committed:
Fixed invalid task created during rebase.

mohammedzee1000 committed:
Specifying specific fields in serializers for api.

mohammedzee1000 committed:
Adding filtering backends to API server.

mohammedzee1000 committed:
Removing filtering backends in favour of manual filter.

mohammedzee1000 committed:
Implementing manual filtering on Project view.

mohammedzee1000 committed:
Fixing column name in serializer.

mohammedzee1000 committed:
Implementing basic filtering on api endpoint views.

mohammedzee1000 committed:
Adding \n after line continuation in import.

mohammedzee1000 committed:
Adding missing quotes.

mohammedzee1000 committed:
Adding more test cases.

mohammedzee1000 committed:
Fixes pep8 issues with multiple python modules

bamachrn committed:
Fixed all the registry endpoints to get proper value in GUI

mohammedzee1000 committed:
Removing api tests for now.

mohammedzee1000 committed:
Fixing combined use of become and sudo.

mohammedzee1000 committed:
Adding target file link field to database.

mohammedzee1000 committed:
Updating query string param to name for refering project.

This is done so project name in projects table matches
name in Builds table.

mohammedzee1000 committed:
Fixing all sudos in django tasks.

navidshaikh committed:
Refactors base class for scanners and adds utilities

  The new refactored base class now has organized utilities for scanning.

navidshaikh committed:
Refactors individual scan invokers and scan runner

  also removed output file names config altogether from settings.

mohammedzee1000 committed:
Updaing import expression for constants.

navidshaikh committed:
Updates scanner functional tests to use individual invoker classes

   Now result file name is used from individual invoker classes.
   Enabled pipeline scanner result file test.
   Reduced amount of sleep time while expecting results to appear.

navidshaikh committed:
Fixes retrieving image identifier key name from job data

navidshaikh committed:
Unifies scanner variable names across invoker classes

navidshaikh committed:
Updates base and subsclass scan and run methods

  Also fixes scnanner_name --> scanner variable

navidshaikh committed:
Adds missing mount point to mount object for pipeline scanner

navidshaikh committed:
Fixes processing result on failure for pipeline scanner

navidshaikh committed:
Fixes processing output for misc-package-updates scanner

navidshaikh committed:
Corrects the installed name of rpm-verify scanner

navidshaikh committed:
Exports scanner execution status file with indent and sorted keys

navidshaikh committed:
Fixes processing output result format for pipeline scanner

navidshaikh committed:
Sets required env vars in base class instead of subclass

navidshaikh committed:
Improves default result processing of base class

navidshaikh committed:
Fixes imports in scanner functional tests

  Using the result file names of each scanner as fixtures and avoiding
  to import individual scanner invoker class in tests.

mohammedzee1000 committed:
Removing files carried over from different branch.

navidshaikh committed:
Show only image name in scanner output

  Earlier registry name along with random tag used scanning will
  show up in scanners output. This changeset adds a utility method
  to split the repository_name and only add image_name without tag
  in scanners output.

dharmit committed:
Fix settings for scanner worker to look for correct file

navidshaikh committed:
Merge pull request #570 from dharmit/fix-569

Fix settings for scanner worker to look for correct file

navidshaikh committed:
Uses devicemapper storage option for devcloud environments

  Fixes #567: We were using default storage driver overlay2 on devcloud
  environment, causing non writable mounts for pipeline-scanner.
  This changeset updates provisioning scripts to use devicemapper as
  storage driver when deployment environment is not in [production, pre-prod, staging].

bamachrn committed:
Merge pull request #372 from mohammedzee1000/11-23-2017_14-00-18_pipeline_data_rest_api

Adding api to the container pipeline service data.

mohammedzee1000 committed:
Fixing target file link addition in weekly scan.

Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

navidshaikh committed:
Stops using separate project to check scanners results, use existing

  Now scanners tests will check for scanners results of projects
  built already via openshift tests.
  This simplifies the tests and overall running time.

mohammedzee1000 committed:
Removing unused import statement.

navidshaikh committed:
Merge pull request #573 from mohammedzee1000/04-16-2018_12-10-21_fix_weekly_scan_db_gc

Fixing target file link addition in weekly scan.

dharmit committed:
Merge pull request #571 from navidshaikh/fix-devicemapper

Uses devicemapper storage driver for devcloud environments

bamachrn committed:
Update openshift version to fix oc retrive logs in workers

dharmit committed:
Merge pull request #558 from navidshaikh/fix-550

Improves scanning module, better error handling and cleanup, DRY scan invokers

bamachrn committed:
Added port to public registry in ci

bamachrn committed:
Added sample for public registry in inventory sample

bamachrn committed:
Added intranet registry as preferance to pull images

bamachrn committed:
resotred hosts.sampl to make registry ui work

bamachrn committed:
Removing unnecessarry public registry

dharmit committed:
Merge pull request #520 from bamachrn/fix-registry-url

Fixed all the registry endpoints to get proper value in GUI

mohammedzee1000 committed:
Refactoring index CI to a more modular approach.

Index CI is being re-written to allow modularity
and an easier way to use the existing validation.

mohammedzee1000 committed:
Fixing pep8 issues in code.

mohammedzee1000 committed:
Fixed bug with overall success and removed DependsOnValidation.

Depends on validation moved out of this patch.

bamachrn committed:
Removed gitignores to get all the files in repo

bamachrn committed:
Removed gitignores to get all the files in repo

dharmit committed:
Merge pull request #584 from bamachrn/pre-build-git-push

Removed gitignores to get all the files in repo

mohammedzee1000 committed:
Removing state related stuff from PR, and it will be part of pr 2.

mohammedzee1000 committed:
Changing clone function to directly clone to destination.

mohammedzee1000 committed:
Added code to enable storage of state accross run.

mohammedzee1000 committed:
Adding constants for state data.

mohammedzee1000 committed:
Adding state initialization and cleanup to the engine.

mohammedzee1000 committed:
Updating state logic for storage of state and keys.

mohammedzee1000 committed:
Correcting logic for yaml dump and adding 2 more functions.

mohammedzee1000 committed:
Adding base classes for stateful validations.

mohammedzee1000 committed:
Adding UniqueEntryValidator as first stateful validatior.

mohammedzee1000 committed:
Enabling UniqueEntryValidator by default.

mohammedzee1000 committed:
Removed uuid as its no longer needed here and fixed import.

mohammedzee1000 committed:
Adding constants to be used by validators.

mohammedzee1000 committed:
Updating clone function to strip leading //

mohammedzee1000 committed:
Removing unnessasary ".

mohammedzee1000 committed:
Updating and renaming cloned validator to allow all post clone validations.

mohammedzee1000 committed:
Wiring the value validators into the engine.

dharmit committed:
Parameterize container index repo and branch

Filters configuration files during execution of rpm verify scan

Invokes python script from slave container for scan

  A new image is also referenced which contains the mentioned
  python script for yum updates scan.

  In scan stage, the scan container is created with volume share
  of /opt/scanning dir. This dir is present in the slave container.
  This dir contains the yum updates scan script.
  The volume is made available in scan container, and the respective
  script is invoked using docker exec in matching scan stage.

[tmp] Serves image from temp registry until image appears on r.c.o

Removes extra whitespaces while redirecting stderr to stdout

DEBUG: Check the contents of /opt/ dir in slave container

Uses only docker run approach for scan

Overrides the work dir while creating the scan container

DEBUG: Check the docker run with entrypoint approach

mohammedzee1000 committed:
Adding value_validation.

mohammedzee1000 committed:
Adding a print function to utils.

mohammedzee1000 committed:
Fixing engine.py based on review.

- Adding error message and flow in case we failed to fetch data from index file.
- Adding docstrings.
- Fixing imports

mohammedzee1000 committed:
Updating utils as per review by navidshaikh

- Fixing imports
- Moved return True out of exception
- load and dump yaml now returns error message that can be used by callee.

mohammedzee1000 committed:
Fixing load_yaml call to handle err and display it to user.

mohammedzee1000 committed:
Fixing basevalidation.py as per comments by navidshaikh

- Adding missing docstrings.
- Adding default value UNKNOWN to field_name.
- Fixing typo.

mohammedzee1000 committed:
Fixing schema_validation as per comments by navidshaikh.

- Adding docstrings
- Changing dicts to dictionaries
- Fixing consistency of Id im comments

mohammedzee1000 committed:
Fixing pep8 issues.

navidshaikh committed:
Merge pull request #582 from mohammedzee1000/index_ci_refac

Refactoring index CI to a more modular approach.

mohammedzee1000 committed:
Fixing pep8 issues.

mohammedzee1000 committed:
Fixing issues based on comments by @navidshaikh

- Fixing all imports
- Adding missing docstrings
- Fixed formatting issues with code.

mohammedzee1000 committed:
Adding Pre-build schema validation.

mohammedzee1000 committed:
Adding validation for target-file.

mohammedzee1000 committed:
Adding CCCP Yaml Validator.

mohammedzee1000 committed:
Adding new validations to the index ci.

- Updating CCCP Exists Validator to use new CCCPValidator base.
- Adding prebuild exists and Job ID matching validation.
- Target file validation now skips if prebuild is present.

mohammedzee1000 committed:
Adding newly added validations to validation list.

mohammedzee1000 committed:
Fixing pep8 issues.

mohammedzee1000 committed:
Adding missing docstrings and comments.

mohammedzee1000 committed:
Fixing space in comment.

mohammedzee1000 committed:
Adding tests for new index CI.

- Adding basic framework.
- Adding tests for projects validation and id validation.

navidshaikh committed:
Merge pull request #585 from mohammedzee1000/index_ci_refac_2

This is the 2nd patch of the index ci refactor series.

mohammedzee1000 committed:
Fixing test order and adding more tests.

mohammedzee1000 committed:
Adding missing docstrings.

mohammedzee1000 committed:
Adding few more schema validation tests and fixing some exisiting ones.

mohammedzee1000 committed:
Adding build context schema validatation.

mohammedzee1000 committed:
Adding pre-build validaion tests and fixing issues found in validator.

mohammedzee1000 committed:
Altered the way state is implemented.

State can now be maintained by each callee
individually and people can juggle around
state objects.

mohammedzee1000 committed:
Adding variable to allow state to be sent as validation data.

mohammedzee1000 committed:
Altered validators and engine to use new state format.

Stateful validator updated to read state from validation data
Engine initialized state object to share with the validators.

navidshaikh committed:
Merge pull request #586 from mohammedzee1000/index_ci_refac_3

Adding remaining baseline validations to index ci

navidshaikh committed:
Adds documentation about configuring DaemonSet

mohammedzee1000 committed:
Removing unnessasary functions.

mohammedzee1000 committed:
Adding UniqueEntryValidation tests.

mohammedzee1000 committed:
Adding tests for git clone validation.

mohammedzee1000 committed:
Fixing pep8 issues.

navidshaikh committed:
Fixes validating projects against db from index in weekly scan

  We must validate project entry db against entry in index before
  putting the weekly scan job on tube.
  Handles the DoesNotExist exception raised by django, if entry in index
  does not match in projects table of db.

navidshaikh committed:
Merge pull request #588 from mohammedzee1000/index_ci_state_update

Provisions having separate instance of state object

mohammedzee1000 committed:
Updating state to be able to handle environment for git.

mohammedzee1000 committed:
Adding environment setup to OptionalCloneValidator as it does git clone.

mohammedzee1000 committed:
Adding remaining validations for git clone.

mohammedzee1000 committed:
Altering how CI index CI works with paths containing gitpath.

Git path can start with "/" which is problem for os.path.join

mohammedzee1000 committed:
Adding a mock directory to state for tests and any other rough items.

mohammedzee1000 committed:
Adding base class with common function for file exists validations.

mohammedzee1000 committed:
Adding tests for cccp yaml exists validations.

mohammedzee1000 committed:
Adding tests for validating TargetFile validation.

mohammedzee1000 committed:
Renaming FileBaseTest, adding PreBuildValidation tests, and fixing.

mohammedzee1000 committed:
Adding tests for job id in cccp yaml validation and fixing issues.

mohammedzee1000 committed:
Removing unnessasary function.

mohammedzee1000 committed:
Fixing <= to ==.

mohammedzee1000 committed:
Renamed functions that had missed rename in git url schema tests.

mohammedzee1000 committed:
Fixing names in git path validation.

navidshaikh committed:
Sends email for weekly scan only if scanners execution status=true

  In case of scanners failed to execute, prevent empty emails to be sent.
  Also corrects the build status to `failed` if scanners failed to execute.

navidshaikh committed:
Merge pull request #590 from CentOS/fix-589

Fixes validating projects against db from index in weekly scan

mohammedzee1000 committed:
Updating container index ci engine to allow state to be passed.

The engine can now be given a state object to use. This enables
user to create the state object outside, initialize extra stuff,
and then pass it to the engine to use.

Also adding functional tests.

navidshaikh committed:
Merge pull request #587 from mohammedzee1000/index_ci_refac_4

Adding tests for container index CI

mohammedzee1000 committed:
Engine can now skip validations, by specifying None.

Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

mohammedzee1000 committed:
Removed unused imports.

mohammedzee1000 committed:
Adding functional tests for index ci and removing debug statements.

mohammedzee1000 committed:
Changed the way to to specify no validations.

mohammedzee1000 committed:
Adding return if cccp yaml data failed to load.

mohammedzee1000 committed:
Updating tests to use the new way of no validators.

mohammedzee1000 committed:
Adding ability to manipulate parsers to init.

mohammedzee1000 committed:
Adding index ci functional tests for command line invokation.

The tests will now need to be invoked with sudo privilages.

mohammedzee1000 committed:
Changing toe check_call.

mohammedzee1000 committed:
Rewiring index ci to run the new validations.

bamachrn committed:
Added jenkins file for ci-functional test

bamachrn committed:
Added job template for CI

bamachrn committed:
Added ci trigger point

mohammedzee1000 committed:
Removing references that pull different branch.

mohammedzee1000 committed:
Removing pythonpath as it can be added while calling the code.

mohammedzee1000 committed:
Altering script to use python path.

navidshaikh committed:
Merge pull request #591 from mohammedzee1000/index_ci_refac_5

Adding more tests for the index CI

mohammedzee1000 committed:
Fixing the way you can skip entire set of tests.

mohammedzee1000 committed:
Fixing the parameters according to changes in above commit.

mohammedzee1000 committed:
Fixing the tests to work with new way of skipping validations.

mohammedzee1000 committed:
Fixed based om comments by @navidshaikh

bamachrn committed:
Added approvals for running the JSON checks

bamachrn committed:
Added function to remove repeated functionalities

mohammedzee1000 committed:
Setting pythonpath before running the CI.

The pythonpath needs to be set to root of service code
or imports will fail.

bamachrn committed:
Added parallel check for both success and failure

bamachrn committed:
Added assertions for success and fail cases

mohammedzee1000 committed:
Updating yaml loader to use base loader.

navidshaikh committed:
Merge pull request #595 from mohammedzee1000/07-17-2018_14-42-38_pipeline_ci_python_path

Setting pythonpath before running the CI.

mohammedzee1000 committed:
Removing yaml loader, setting it to default.

mohammedzee1000 committed:
Fixing the help test for parameter.

mohammedzee1000 committed:
Removing unneeded commas in test data.

bamachrn committed:
Added nfs setup provisions

mohammedzee1000 committed:
Set python path for ci run.

mohammedzee1000 committed:
Adding debug messages.

bamachrn committed:
Merge pull request #593 from mohammedzee1000/index_ci_refac_6

Adding init for new index ci and adding tests for the same

bamachrn committed:
Merge pull request #594 from mohammedzee1000/index_ci_refac_rewire_index_ci

Rewiring index ci to run the new validations.

mohammedzee1000 committed:
Adding more explicit setting of pythonpath for index ci.

navidshaikh committed:
Merge pull request #596 from mohammedzee1000/index_ci_pythonpath

Adding more explicit setting of pythonpath for index ci.

mohammedzee1000 committed:
Fixing validator load conditions.

Still need to figure out how this slipped pass.

mohammedzee1000 committed:
Fixing index ci to skip cccp yaml validation if prebuild is specified.

mohammedzee1000 committed:
Fixing the print function to not print state.

bamachrn committed:
Added ansible-controller

bamachrn committed:
Added common for all

bamachrn committed:
Allow all requests from the cluster nodes

mohammedzee1000 committed:
Adding logic to skip template file.

mohammedzee1000 committed:
Fixing prebuild validation to check prebuild script relative to context.

mohammedzee1000 committed:
Fixing index ci tests for pre-build validation.

bamachrn committed:
Fixed nfs export for PV

bamachrn committed:
Added Docker distribution to NFS node

bamachrn committed:
Deploy Openshift cluster

mohammedzee1000 committed:
Temporarily adding skip for centos base image.

bamachrn committed:
Deploy jenkins with persistent volume

bamachrn committed:
Added script approval for running json checks

mohammedzee1000 committed:
Disabling stringfield validation on desired tag.

mohammedzee1000 committed:
Adding library images to list of those where job id match is skipped.

mohammedzee1000 committed:
Adding few exceptions to index ci until they can be sorted.

bamachrn committed:
Merge pull request #601 from mohammedzee1000/index_ci_fixes

Adding few exceptions to index ci until they can be sorted.

mohammedzee1000 committed:
Making the skip of fabric8-ui images more loose.

bamachrn committed:
Merge pull request #602 from mohammedzee1000/index_ci_fixes

Making the skip of fabric8-ui images more loose.

mohammedzee1000 committed:
Adding fabric8 namespace as well.

bamachrn committed:
Added hosts file for OpenShift 3.9

bamachrn committed:
Added ci trigger on CICO

bamachrn committed:
Added trigger script for CI

bamachrn committed:
Set log level to error

mohammedzee1000 committed:
Removing old index ci in favour of new one.

This PR removes the old index ci along with tests associated
with the old index ci.

bamachrn committed:
Changed ci index repo to upstream

bamachrn committed:
Run CI pipeline and release nodes after done

bamachrn committed:
Move build config to use scm plugin instead of git

bamachrn committed:
Added openshift authentication

navidshaikh committed:
Merge pull request #604 from mohammedzee1000/delete_old_index_ci

Removing old index ci in favour of new one.

bamachrn committed:
Fetch CI logs

navidshaikh committed:
Adds script to run unittests in Centos CI infrastructure

 This patch adds script which invokes CI job at cico for
 running unittests.

bamachrn committed:
Brought jenkins file to inline

bamachrn committed:
Added pre-requisite for deployment

bamachrn committed:
Wait for seed job to complete before ci starts

bamachrn committed:
Merge branch 'openshift' of github.com:bamachrn/ccp-openshift into openshift

bamachrn committed:
Removed old code base in faovour of new one

bamachrn committed:
Separate success job

bamachrn committed:
Added failure check pipeline

bamachrn committed:
Functional CI work flow diagram

This diagram explains the flow of functional CI

bamachrn committed:
Added documentation for functional tests

navidshaikh committed:
Converts pipeline name to lower case

 Fix OSIO 1522

 Since OpenShift doesn't allow the value of `metadata.name` to have uppercase chars.

 This commit also adds a unit test case.

dharmit commented on ccp/index_reader.py:
```diff
@@ -240,7 +242,7 @@ def apply_build_job(self,
         Applies the build job template that creates pipeline to build
         image, and trigger first time build as well.
-        :param project: The name of project, where the template is to be applied
+        :param project: The name of project,where the template is to be applied
```
This space is removed by error?

navidshaikh replied:
No, it was removed to fit the length of the string < 80 chars.

navidshaikh replied:
Fixed.

dharmit reviewed (commented):
LGTM except minor bit

navidshaikh committed:
Ensures the pipeline name confronts to allowed pattern

 Added a regular expression to validate the pipeline name generated
 for using in template's metadata.name field. oc imposes certain guidelines
 for same and provides the regular expression in error/help message to be
 used for validation in case it found some invalid string.

 We are using same regular expression as provided by oc to validate
 the pipeline name while processing index and generating build configs
 from it.

 Added a custom exception, which will be raised if the populated string
 is invalid.

 Added a unit test as well to test the functionality.

navidshaikh commented:
Added a regular expression to validate the pipeline name generated
 for using in template's metadata.name field. oc imposes certain guidelines
 for same and provides the regular expression in error/help message to be
 used for validation in case it found some invalid string.

 We are using same regular expression as provided by oc to validate
 the pipeline name while processing index and generating build configs
 from it.

 Added a custom exception, which will be raised if the populated string
 is invalid.

 Added a unit test as well to test the functionality.

navidshaikh commented:
All the other notification PRs need to be rebased onto this.

bamachrn committed:
Getting CI logs for both success and failure

bamachrn committed:
Updated README with both success and failure logs

bamachrn committed:
Added comments to both pipelines

dharmit committed:
Merge pull request #608 from bamachrn/openshift

OpenShift based new architecture and its functional CI

bamachrn committed:
Merge pull request #613 from navidshaikh/os-unittests

Adds script to run unit tests in Centos CI infrastructure
